pkgs/build-support/setup-hooks/make-wrapper.sh

```diff
@@ -1,3 +1,12 @@
+# Assert that FILE exists and is executable
+#
+# assertExecutable FILE
+assertExecutable() {
```

Comment:
It is handled properly.  Cases I just tested:

  * non-executable file: Assertion failure
  * symlink to non-executable file: Assertion failure
  * executable file: Success
  * symlink to executable file: Success
  * traversable directory: Assertion failure
  * symlink to traversable directory: Assertion failure